ProjectCoimbra.UWP/Project.Coimbra/Helpers/SongPagesHelper.cs

```diff
@@ -185,5 +184,31 @@ public static async Task FillListBoxAsync(ListBox listBox, string selectedValue,
                 nextButton.IsEnabled = true;
             }
         }
+
+        /// <summary>
+        /// Removes Item from list
+        /// </summary>
+        /// <param name="key">item key</param>
+        public static async Task RemoveFile(string key)
+        {
+            if (string.IsNullOrWhiteSpace(key))
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            var midiFolder = await GetMidiFolderAsync(false).ConfigureAwait(true);
+            if (midiFolder == null)
+            {
+                throw new ArgumentNullException(nameof(midiFolder));
```

Comment:
It seems that returning `Task<bool>` would be better here. Return values are more efficient than exceptions and as such exceptions should only be thrown for truly exceptional cases.

Also, `ArgumentNullException` isn't the correct exception here. Most code validators would reject this, so if you're going with exceptions, we should get something more appropriate.